### DISCUSSION TITLE
Batch read prompts from a folder.

### DISCUSSION THREAD

**Discussion by nykwil** [receyuki/comfyui-prompt-reader-node#22] [Q&A] [answered]
How would I hook it up so that I can read a bunch of files and iterate through them using a batch function. Like if there was a filename pin that I could load a folder directly from or if you could do it directly from the node. I noticed you can break out the filename and it lets you iterate through the list but how do I add files to that list?

**Comment by receyuki** [comment]
Duplicate #13
I will add this feature in v1.1.0.

**Comment by metamountain** [comment]
I would also love that !!!!

**Reply by metamountain** [reply]
It would be fine ,if you could just wire an image as input for the promptreader. (Source might be a batch upload)

**Reply by receyuki** [reply, marked as answer]
Please check this out
https://github.com/receyuki/comfyui-prompt-reader-node#batch-loader-node

**Reply by metamountain** [reply]
Incremental batch still not successful. Perhaps you can implement the batchloader in the example .json file.   Thanks in Advance ! Great work !

**Reply by metamountain** [reply]
Seems to be broken because of updates...

**Reply by receyuki** [reply]
Perhaps next time you could post your question in the issues rather than in the discussions, I'm not sure why I don't receive notifications for discussions.
About the batch loader, are you able to use it now? If not, you can post screenshots and the workflow to the issues, so I can determine where the problem lies.

**Comment by Jonseed** [comment]
@receyuki How do you use the Batch Loader node to iterate through images in a folder, one queue per image? I can't figure out how to use it. It seems to send all the images through at once.

**Reply by Jonseed** [reply]
Actually, it looks like it is working. It seems like it wasn't because some nodes process all the images at once, and other nodes are sequential. I thought it would go through the whole queue for each image, but that isn't how it works. It sends all the images through each node before proceeding to the next node.

**Reply by Jonseed** [reply]
Although it would help workflows if it was possible to have Batch Loader just send through 1 image per queue. Is there a way to have it do that?

**Comment by nykwil** [comment]
You can get an incrementer then iterate through the start_index with max
loaded set to 1

**Reply by Jonseed** [reply]
Where would I find "max loaded"?

**Comment by nykwil** [comment]
image_load_limit, come on, just look at it for a bit before you ask
questions.

**Reply by Jonseed** [reply]
not helpful
